universe/interface.go

```diff
@@ -571,6 +571,19 @@ const (
 	ProofTypeTransfer
 )
 
+// NewProofTypeFromAssetProof returns the proof type for the given asset proof.
+func NewProofTypeFromAssetProof(proof *proof.Proof) (ProofType, error) {
+	if proof == nil {
+		return ProofTypeUnspecified, fmt.Errorf("proof is nil")
+	}
+
+	if proof.Asset.HasGenesisWitness() {
```

Comment:
Also needs `HasGenesisWitnessForGroup` as well. I think we should have some wrapper function/method to unify them so we don't always forget.